src/components/common/EdgeAnim.tsx

```diff
@@ -135,4 +140,12 @@ const EdgeAnimInner = ({ children, disableAnimation, enter, exit, visible = true
   )
 }
 
-export const EdgeAnim = React.memo(EdgeAnimInner)
+const edgeAnimPropsAreEqual = (prevProps: Props, nextProps: Props): boolean => {
+  const { children: prevChildren, ...prevRest } = prevProps
```

Comment:
I would also short cut check each prop is === to each other and return true if so. Only if they are not equal AND the props that are not === are Objects do you use matchJson